api/src/main/java/com/exadel/frs/core/trainservice/component/classifiers/EuclideanDistanceClassifier.java

```diff
@@ -83,9 +84,12 @@ private INDArray normalize(final INDArray embeddings) {
 
     private double[] recognize(final INDArray newFace, final INDArray existingFaces) {
         val distance = euclidean_distance(newFace, existingFaces);
-        distance.divi(2).rsubi(1);
 
-        return distance.toDoubleVector();
+        return calculateSimilarities(distance).toDoubleVector();
+    }
+
+    private INDArray calculateSimilarities(INDArray distance) {
+        return tanh(distance.rsubi(1.1).muli(2.5)).addi(1).divi(2);
```

Comment:
Please use in-place tanh method
```suggestion
        return tanh(distance.rsubi(1.1).muli(2.5), false).addi(1).divi(2);
```